### PR TITLE
feat(hot-reload): introduced postSyncCommand

### PR DESCRIPTION
**Pull request by thsig**
Adds a new config option to hot reload specs: `postSyncCommand`.

This is a command that is run inside the container after a hot reload sync takes place.

Also added a example project to demonstrate how this feature might be useful.

Note: Didn't link-ify references to the `hot-reload-post-sync-command` example project, since I didn't want to trip up the link validator. Should I just be using relative links here?

### CODE REVIEW

File: examples/hot-reload-post-sync-command/garden.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+kind: Project
+name: hot-reload-callback
```
Comment (edvald) [inline examples/hot-reload-post-sync-command/garden.yml]: Name should probably match the directory name. Also, could you make a note to add this to our e2e tests when #1091 is done?

File: examples/hot-reload/README.md [hunk 2]
```diff
@@ -16,7 +16,7 @@ In the `garden.yml` file of the `node-service` module we first enable hot-reload
 ...
 hotReload:
   sync:
-    - target: /app/
+
```
Comment (edvald) [inline examples/hot-reload/README.md]: Is this intentional?

File: examples/hot-reload-post-sync-command/node-service/package.json [hunk 3]
```diff
@@ -0,0 +1,16 @@
+{
+  "name": "good-morning",
+  "version": "1.0.0",
+  "description": "Good morning greeting service",
```
Comment (edvald) [inline examples/hot-reload-post-sync-command/node-service/package.json]: This service should be renamed, no?

File: examples/hot-reload-post-sync-command/node-service/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,12 @@
+FROM node:9-alpine
+
+ENV PORT=8080
+EXPOSE ${PORT}
+
+ADD . /app
+WORKDIR /app
+
+RUN npm install -g nodemon
```
Comment (edvald) [inline examples/hot-reload-post-sync-command/node-service/Dockerfile]: This line could be above the above `ADD`

**Comment by edvald** [comment]
Just some small comments, otherwise 👍 

Oh and I'd suggest making the commit category `feat(container)` since this is specific to that module type.

File: examples/hot-reload/README.md (hunk 2 above)
Comment (thsig) [reply examples/hot-reload/README.md]: Nope, good catch.

File: examples/hot-reload-post-sync-command/node-service/package.json (hunk 3 above)
Comment (thsig) [reply examples/hot-reload-post-sync-command/node-service/package.json]: Yeah, it should. I'll perform the same rename in the `hot-reload` example project.

**Comment by thsig** [comment]
I'll rebase this on `master` once https://github.com/garden-io/garden/pull/1213 has been merged.

**Pull request merged**